src/include/wiredtiger_ext.h

```diff
@@ -532,6 +532,27 @@ struct __wt_extension_api {
     int (*transaction_visible)(
       WT_EXTENSION_API *wt_api, WT_SESSION *session, uint64_t transaction_id);
 
+    /*!
+     * Initialize a spinlock
+     */
+    int (*spin_init)(
+      WT_EXTENSION_API *wt_api, WT_SESSION *session, void *spinlock, const char *name);
```

Comment:
I'd rather not pass `void` pointers across the API boundary. Could you create a dummy type for the spinlock in here, and cast to/from it in the API implementation functions?

e.g:

```
typedef struct __wt_extension_spinlock WT_EXTENSION_SPINLOCK;
/*!
 * A placeholder data structure that allows for using the WiredTiger
 * spinlock implementation from within extensions.
 */
struct __wt_extension_spinlock {
};
```

I'm also a little surprised by the level of indirection on the spinlock pointer - should it be a **?

Reply:
Instead of duplicating the spinlock structure, I am planning to have 
`/*!
 * A placeholder data structure that allows for using the WiredTiger
 * spinlock implementation from within extensions.
 */
struct __wt_extension_spinlock {
        void *lock;   /* spinlock */
}`

Reply:
Sounds good to me.